src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -371,9 +371,7 @@ def module_is_sequentially_offloaded(module):
             if not is_accelerate_available() or is_accelerate_version("<", "0.14.0"):
                 return False
 
-            return hasattr(module, "_hf_hook") and not isinstance(
-                module._hf_hook, (accelerate.hooks.CpuOffload, accelerate.hooks.AlignDevicesHook)
-            )
+            return hasattr(module, "_hf_hook") and isinstance(module._hf_hook, accelerate.hooks.AlignDevicesHook)
```

Comment:
`enable_sequential_cpu_offload` install `AlignDevicesHook` so I think it is a mistake here - it did not affect the results because when this type of hooks is installed, you would not be able to move the models to other devices anyway 

i.e., 
with this change, this script will now throw the `ValueError` we intended to throw. 

Before this change, you would get a `NotImplementedError: Cannot copy out of meta tensor; no data! ` when it tried to move the models to gpu

```python
from diffusers import StableDiffusionPipeline
import torch
from accelerate.hooks import CpuOffload, AlignDevicesHook, remove_hook_from_module

num_inference_steps = 2
pipe = StableDiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-2-1")
pipe.enable_sequential_cpu_offload()
pipe.to("cuda")
```